Fix init_process not preventing processing

LeDrascol opened the pull request:
## About The Pull Request
Prevents machines from running faster when set to both **not** init_process and **not** speed_process. Changes mineral machine init_process to use this fix.

Related to a Sandstorm machine issue here: https://github.com/Sandstorm-Station/Sandstorm-Station-13/pull/276

## Why It's Good For The Game
Setting a machine to not start when constructed should actually do that.

## Changelog
:cl:
fix: Fixed non-autostart machines autostarting
/:cl:

LeDrascol committed:
Fix init_process not preventing processing

Prevents machines from running faster when set to both not init_process and not speed_process.

Changes mineral machine to init_process to fix potential issues caused by this.

SandPoot commented on code/modules/mining/machine_processing.dm:
```diff
@@ -4,7 +4,7 @@
 /obj/machinery/mineral
 	speed_process = TRUE
-	init_process = FALSE
+	init_process = TRUE // Must be true after init_process fix
```
I really think this shouldn't be touched

LeDrascol replied:
This is a required change to prevent mineral machines from no longer processing. The current machines require `speed_process = TRUE` and `init_process = FALSE` due to an error that is being resolved.

SandPoot commented on code/modules/mining/machine_processing.dm:
```diff
@@ -4,7 +4,7 @@
 /obj/machinery/mineral
 	speed_process = TRUE
-	init_process = FALSE
+	init_process = TRUE // Must be true after init_process fix
```
```suggestion
	init_process = FALSE
```
leave it

LeDrascol committed:
Merge branch 'master' into fix-machine-init-process

LeDrascol committed:
Selective mineral machine init_process

Requested by SandPoot. Disables init_process for mineral (mining) machines, and enables it on a case-by-case basis.

Downstream mineral machines must be updated with this change.

LeDrascol committed:
Move furnace init variable

Moves the init_process variable above the var defines to drastically improve style.

Requested by SandPoot.
